### PR TITLE
FEATURE: RAIL-1197 Implement menu dropdown component

### PR DESCRIPTION
**Pull request by Hurtak**
Adds menu to pivot table. With menu we can now turn on/off totals in the pivot table. Pivot table now uses hybrid approach that takes props of totals, saves them to its state, and uses the state to render the totals. Whenever menu totals are changed from the menu, the state is updated, whenever props change, the totals state is replaced with new totals props.

Caveats of current implementation:
- In pivot table dynamic examples there is settings of the pivot table, this settings (the totals part) is not updated when we change totals from within the menu.
- pivot table apparently uses preventDefault on scroll events, so menu is using `addEventListener("scroll", document, true)`, useCapture: true is used to catch this event from the top->bottom direction of event bubbling, before preventDefault can be called. And because menu catches the scroll event before the pivot table, it sometimes leads to outdated position (off by 1 px usually, so not that noticeable) of menu relative to the pivot table cell (because menu is positioned **first** and pivot table content is positioned **after**). This is only problem for menu when it deals with elements that are positioned with javascript that are reacting to scroll event. Easiest solution that comes to mind could be that menu would update its position not directly after scroll event happens, but after `setTimeout(fn, 0)`, but this could lead to janky experience of menu position because its position would be update in next browser frame instead the current one. We could update the position the current frame **and** the next frame, but this seemed like lots of unnecessary computations that only solve this very specific problem that is not that noticeable.
- HeaderCell tests for menu presence are deleted because menu now requires executionResponse that would need to be mocked and in order to display it, it requires hover events that are better tested in integration tests anyway. So I decided to remove these tests and cover these cases in Testcafe tests.

### CODE REVIEW

**Comment by Hurtak** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2608/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2007/ : FAILURE in 16m 15s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1067/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2609/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/624/ : UNSTABLE in 23m 24s

**Comment by Hurtak** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1069/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2611/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/625/ : UNSTABLE in 17m 49s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1070/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2613/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/626/ : UNSTABLE in 19m 28s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by Hurtak** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1074/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2629/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/631/ : UNSTABLE in 36m 49s

**Comment by Hurtak** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1075/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2632/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/633/ : SUCCESS in 20m 54s

File: src/components/core/PivotTable.tsx [hunk 1]
```diff
@@ -251,21 +254,23 @@ export const getGridDataSource = (
...
     getGridApi: () => any,
     intl: InjectedIntl,
-    columnDefOptions: IColumnDefOptions = {}
+    columnDefOptions: IColumnDefOptions = {},
+    columnTotals?: AFM.ITotalItem[]
 ): IDatasource => ({
     getRows: ({ startRow, endRow, successCallback, sortModel }: IGetRowsParams) => {
         const execution = getExecution();
+
+        const resultSpecFinal: AFM.IResultSpec = resultSpec;
```
Comment (tomasreichmann) [inline src/components/core/PivotTable.tsx]: {...resultSpec}

Comment (tomasreichmann) [reply src/components/core/PivotTable.tsx]: https://petapixel.com/2015/07/22/this-is-how-everyone-names-their-final-psd-files/

Comment (Hurtak) [reply src/components/core/PivotTable.tsx]: Updated

**Comment by Hurtak** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Github ref error: SHA of pull request head checked out from remote PR head ref is different from head commit SHA from Github webhook event.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by Hurtak** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2639/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2029/ : SUCCESS in 15m 51s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1079/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2640/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/636/ : UNSTABLE in 22m 53s

File: src/components/core/pivotTable/HeaderCell.tsx [hunk 2]
```diff
@@ -63,70 +69,176 @@ export default class HeaderCell extends React.Component<IProps, IState> {
...
+                ) as PivotTableMenuTotalType[];
+        } else if (lastFieldType === 'a') {
+            const isColumnAttribute = lastFieldValudId === null;
+            if (isColumnAttribute) {
+                return null;
+            }
+
+            localIdentifiers = measureGroupHeaderItems.map(i => i.measureHeaderItem.localIdentifier);
+            turnedOnTotals = supportedTotalTypes
+                .filter((type) => {
```
Comment (tomasreichmann) [inline src/components/core/pivotTable/HeaderCell.tsx]: suppertedTotals are filtered in both if branches. Can be moved below both ifs

File: src/components/core/pivotTable/HeaderCell.tsx [hunk 3]
```diff
@@ -63,70 +69,176 @@ export default class HeaderCell extends React.Component<IProps, IState> {
...
+                                <path d="M0 0h8v2H0V0zm0 4h8v2H0V4zm0 4h8v2H0V8z" fill="currentColor" />
+                            </g>
+                        </svg>
+                    </div>
+                }
+                opened={this.state.isMenuOpen}
+                onOpenedChange={this.handleMenuOpenedChange}
+            >
+                <ItemsWrapper>
+                    <Header>AGGREGATE</Header>
```
Comment (tomasreichmann) [inline src/components/core/pivotTable/HeaderCell.tsx]: we should use intl formatMessage here

**Comment by Hurtak** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1080/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2641/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/637/ : SUCCESS in 29m 17s

File: src/components/core/pivotTable/HeaderCell.tsx (hunk 3 above)
Comment (Hurtak) [reply src/components/core/pivotTable/HeaderCell.tsx]: Added

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by Hurtak** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2644/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2032/ : SUCCESS in 5m 02s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1081/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2645/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/638/ : SUCCESS in 24m 47s

**Comment by tomasreichmann** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/575/ : SUCCESS in 27s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2034/ : SUCCESS in 7m 50s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1083/ : SUCCESS in 28m 19s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/545/ : SUCCESS in 28m 22s

**Pull request merged**